arrows/core/uv_unwrap_mesh.cxx

```diff
@@ -0,0 +1,279 @@
+  double height;
+  double width;
+};
+}
+
+namespace kwiver {
+namespace arrows {
+namespace core {
+
+/// Private implementation class
+class uv_unwrap_mesh::priv
+{
+public:
+  /// Constructor
+  priv() : spacing(0.005)
+  {
+  }
+
+  priv(const priv& other)
+  {
+  }
+
+  double spacing;
+};
+
+// Constructor
+uv_unwrap_mesh::uv_unwrap_mesh()
+  : d_(new priv)
+{
+  attach_logger( "arrows.core.uv_unwrap_mesh" );
+}
+
+// Copy constructor
+uv_unwrap_mesh
+::uv_unwrap_mesh(const uv_unwrap_mesh& other)
+  : d_(new priv(*other.d_))
+{
+}
+
+
+// Destructor
+uv_unwrap_mesh
+::~uv_unwrap_mesh()
+{
+}
+
+
+// Get this algorithm's \link vital::config_block configuration block \endlink
+vital::config_block_sptr
+uv_unwrap_mesh::get_configuration() const
+{
+  vital::config_block_sptr config =
+      vital::algo::uv_unwrap_mesh::get_configuration();
+   config->set_value("spacing", d_->spacing,
+                     "Spacing between triangles. It is a percentage of the texture size "
+                     "and should be relatively small (default is 0.005).");
+  return config;
+}
+
+
+// Set the configuration
+void uv_unwrap_mesh
+::set_configuration(vital::config_block_sptr in_config)
+{
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config(in_config);
+  d_->spacing = config->get_value<double>("spacing", d_->spacing);
+}
+
+
+// Check that the algorithm's configuration vital::config_block is valid
+bool uv_unwrap_mesh
+::check_configuration(vital::config_block_sptr config) const
+{
+  double spacing = config->get_value<double>("spacing", d_->spacing);
+  if( spacing <= 0.0 || spacing > 1.0 )
+  {
+    LOG_ERROR( logger(),
+             "spacing parameter is " << spacing << ", needs to be in (0.0, 1.0].");
+    return false;
+  }
+  return true;
+}
+
+
+// Unwrap a mesh
+void uv_unwrap_mesh::unwrap(kwiver::vital::mesh_sptr mesh) const
+{
+  if (mesh->faces().regularity() != 3)
+  {
+    VITAL_THROW( algorithm_exception, this->type_name(), this->impl_name(),
+                 "This algorithm expects a regular mesh with triangular faces.");
+  }
+
+  const mesh_face_array_base& faces = mesh->faces();
+  const mesh_vertex_array<3>& vertices = dynamic_cast<const mesh_vertex_array<3>&>(mesh->vertices());
+
+  // Map each triangle in 2d. The longest edge should be horizontal and its left point is (0, 0)
+  std::vector<triangle_t> triangles(mesh->num_faces());
+  double total_area = 0.0;
+  for (unsigned int f = 0; f < mesh->num_faces(); ++f)
+  {
+    // face 3d points
+    vector_3d pt1 = vertices[faces(f, 0)];
+    vector_3d pt2 = vertices[faces(f, 1)];
+    vector_3d pt3 = vertices[faces(f, 2)];
+
+    // triangle edges
+    vector_3d pt1pt2 = pt2 - pt1;
+    vector_3d pt1pt3 = pt3 - pt1;
+    vector_3d pt2pt3 = pt3 - pt2;
+
+    // find the longest edge and assign it to AB, C is the other point
+    vector_3d AB, AC;
+    int longest_edge;
+    if (pt1pt2.norm() >= pt1pt3.norm() && pt1pt2.norm() >= pt2pt3.norm())
```

Comment:
Optimization suggestion for future work (not to hold up merging this branch):

use `.squaredNorm()` instead of `.norm()` because it avoids the square root and you only need to compare relative values.  Also, you might want to precompute and store these norm values in variables so you don't have to recompute the same value several times.